File: HackerRank/Project_Euler/Project_Euler_31_Coin_sums.py
import math


def Solve(N):

    col = [0 for i in range(N)]
    pounds = [0, 1, 2, 5, 10, 20, 50, 100, 200]
    mem = [col for i in range(9)]

    for i in range(9):
        mem[i][0] = 1


    for row in range(1, 9):

        for col in range(1, N):
            sub = col - pounds[row]
            if sub == 0:
                mem[row][col] = (mem[row - 1][col] + 1) % (10**9 + 7)

            elif (sub > 0):
                mem[row][col] = (mem[row][sub] + mem[row-1][col]) % (10**9 + 7)

            else:
                mem[row][col] = mem[row-1][col] % (10**9 + 7)





    return mem



t = int(input())

N = 10**5+7
list_mem = Solve(N)

for _ in range(t):
    n = int(input())
    n = n % (10**9+7)
    print(list_mem[8][n])

File: HackerRank/Project_Euler/Project_Euler_31_Coin_sums.java
import java.util.*;


class Project_Euler_31_Coin_sums{


    static Scanner sc = new Scanner(System.in);
    //static int ans = 1;


    static public long[][] Solve(int N){

        int[] pounds = {0, 1, 2, 5, 10, 20, 50, 100, 200};
        long[][] mem = new long[9][N+1];
        
        
        for(int i = 0; i < 9; i++)
            mem[i][0] = 1;


        for(int i = 1; i <= N; i++)
            mem[0][i] = 0;

        

        for(int row = 1; row < 9; row++){

            for(int col = 1; col <= N; col++){

                int sub = col - pounds[row];
                //System.out.println(col + " SUB: " + sub);

                if(sub == 0)
                    mem[row][col] = (long)(mem[row-1][col] + 1) % (long)(Math.pow(10, 9) + 7);

                else if(sub > 0){
                    mem[row][col] = (long)(mem[row][sub] + mem[row-1][col]) % (long)(Math.pow(10, 9) + 7);
                }

                else{
                    mem[row][col] = (long)mem[row-1][col] % (long)(Math.pow(10, 9) + 7);
                }

            }
            //System.out.println("-------------------------");

        }

/*
        // ============ READ =============
        for(int row = 0; row < 9; row++){

            for(int col = 0; col <= N; col++){
                System.out.print(mem[row][col] + "\t");
            }
            System.out.println();

        }
        // ============= END - READ ============
*/

        return mem;

    }


    public static void main(String... args){

        int T = sc.nextInt();
        long[][] mem = Solve(10000+1);

        while(T-- > 0){
            
            long N = sc.nextInt();

            N = (int)(N % Math.pow(10,9) + 7);
            
            System.out.println(mem[8][(int)N]);
            

        }
    }

}